{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from utils import uniprotRetrieve\n",
    "from utils import mapping"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1. Download data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To download the data, the Uniprot [REST API](https://www.uniprot.org/help/api%5Fqueries) was used.\n",
    "\n",
    "In a first step, \n",
    "two search queries were used to distinguish between secreted proteins and proteins in the cytoplasm.\n",
    "\n",
    "In a second step, protein sequence similarity (and therefore bias towards some parts in the sequence space) is reduced by mapping the proteins to the UniRef50 groups."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1.1. Construct search Queries"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1.1.1. Cytoplasmic proteins"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* We will limit the dataset to Bacteria.\n",
    "* want to make sure there is some evidence of the protein being in the cytoplasm\n",
    "    - this evidence can be generated by automated pipelines or manually assigned\n",
    "* As an extra saveguard, we search for proteins that have no signal peptide annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "QUERY=\"taxonomy:Gammaproteobacteria (locations:(location:cytoplasm) OR locations:(location:cytosol)) NOT annotation:(type:signal)\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Since we still have to map those proteins to their Uniref50 group (at most 50 percent sequence identity),\n",
    "we only need the identifiers and will not waste bandwith in downloading fasta sequences yet."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "FORMAT=\"fasta\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "FILENAME=\"cytoplasmSearch.fasta\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'cytoplasmSearch.fasta'"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Download the identifiers \n",
    "uniprotRetrieve(FILENAME, format=FORMAT, query=QUERY)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1.1.2 Periplasmic proteins"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* We will limit the dataset to Bacteria. This is however a smaller group as for the cytoplasmic proteins since only gram negative bacteria have a periplasmic space.\n",
    "* want to make sure there is some evidence of the protein being in the periplasm\n",
    "    - this evidence can be generated by automated pipelines or manually assigned\n",
    "* As an extra saveguard, we search for proteins that have a signal peptide annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'periplasmSearch.fasta'"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "QUERY=\"taxonomy:Gammaproteobacteria locations:(location:periplasm) annotation:(type:signal)\"\n",
    "FORMAT=\"fasta\"\n",
    "FILENAME=\"periplasmSearch.fasta\"\n",
    "uniprotRetrieve(FILENAME, format=FORMAT, query=QUERY)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1.2. Use [Uniprot Mapper API](https://www.uniprot.org/help/api_idmapping) to map proteins to their Uniref50"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
